spring-session-jdbc/src/main/java/org/springframework/session/jdbc/JdbcOperationsSessionRepository.java

```diff
@@ -154,6 +157,44 @@
 			+ "WHERE SESSION_ID = ?";
 	// @formatter:on
 
+	/**
+	 * MERGE is SQL standadrd. It is supported by SQL Server and Oracle, and eventually
+	 * other databases.
+	 */
+	// @formatter:off
+	private static final String CREATE_SESSION_ATTRIBUTE_QUERY_MERGE = "MERGE INTO %TABLE_NAME%_ATTRIBUTES x "
```

Comment:
I've got a "bad SQL grammar" exception on Oracle 18c if the SQL query has an ending semicolon.

Reply:
Since this PR was declined, GitHub doesn't seem to be updating it when I push to the branch for it.

I had fixed this a while ago in my branch, so I suggest you grab it from there: https://github.com/candrews/spring-session/tree/attributes-duplicate-key